server/handlers.go

```diff
@@ -280,12 +290,26 @@ func (s *Server) handleConnectorCallback(w http.ResponseWriter, r *http.Request)
 		s.notFound(w, r)
 		return
 	}
+
 	callbackConnector, ok := conn.Connector.(connector.CallbackConnector)
 	if !ok {
 		s.notFound(w, r)
 		return
 	}
 
+	// If using SAML connector, oauthState should be empty
+	// Else, relayState should be empty
+	isSaml := reflect.TypeOf(callbackConnector).String() == "*saml.samlConnector"
```

Comment:
blah. Yeah we should probably add another method to the callback connector interface. For example

```
// State returns the response field with the OAuth2 or SAML state. Allowed values are "state" or "RelayState".
func State() string
```

Then use this here.